### PR TITLE
Choose default dialect form $schema in bowtie validate command

### LINKED ISSUE #995: Make `bowtie validate` default to a detected `$schema` rather than the latest one
After #180 is done, we should make:

```
bowtie validate -i python-jsonschema <(printf '{"$schema": "https://json-schema.org/draft/2019-09/schema"}') <(printf '12')
```

default to Draft 2019-09 rather than 2020-12, by virtue of the present `$schema` property.

In other words, the behavior should be:

* if `-D` is not passed and `$schema` is found in the schema, use that as our dialect
* If `-D` is not passed and `$schema` is not found, default to the latest dialect
* if `-D` is passed, it must match as per #180, or else a warning is raised

### PR DESCRIPTION
**Pull request by sudo-jarvis**
Closes #995.

Configured `bowtie validate` to pick dialect from schema if not specified instead of the latest one. If dialect not specified in schema as well then it would default to the latest one.

----
📚 Documentation preview 📚: https://bowtie-json-schema--1057.org.readthedocs.build/en/1057/

### CODE REVIEW

File: bowtie/_cli.py [hunk 1]
```diff
@@ -715,6 +715,21 @@ def convert(
...
 
 
+def _set_dialect(ctx: click.Context, _, value: _Dialect):
+    """
+    Sets the dialect according to $schema if specified.
+    """
+    if value:
+        return value
+    schema = ctx.params.get("schema")
+    dialect_from_schema = schema.get("$schema") if schema else None
```
Comment (Julian) [inline bowtie/_cli.py:725]: This looks like it will blow up if `schema` is the `True` schema.

Comment (sudo-jarvis) [reply bowtie/_cli.py:725]: Oh yeah you are right!

**Comment by sudo-jarvis** [comment]
@Julian, just to update you currently I am having my exams because of which I am not able to contribute much. I will resume after 3rd april.

**Comment by Julian** [comment]
Merged after a minor tweak to the tests, thanks again!

**Pull request closed without merging**